lib/puppet/provider/ec2_securitygroup/v2.rb

```diff
@@ -105,6 +106,8 @@ def create
       vpc_id = vpc_response.data.vpcs.first.vpc_id
```

Comment:
I think there was one issue here that you were going to address - `vpc_name` doesn't have to be set when you're creating a new `sg` in a VPC only account.  So under those circumstances, (none specified but VPC only account) we have to query for the `vpc_id` that has been attached.